src/renderer/base/renderer.cpp

```diff
@@ -281,9 +281,7 @@ void Renderer::TriggerRedrawCursor(const COORD* const pcoord)
         // The region is clamped within the viewport boundaries and we only
         // trigger a redraw if the region is not empty.
         auto view = _pData->GetViewport();
-        cursorView = view.Clamp(cursorView);
-
-        if (cursorView.IsValid())
+        if (view.IsInBounds(cursorView))

```

Comment:
What a strange way for us to do this. And `IsInBounds` doesn't beef? Neat!

Reply:
I mean the old way was strange.

Reply:
I guess it's a bit late to tell you now, but I don't think this is right. I haven't had a chance to check, but I think `IsInBounds` is testing whether the whole `cursorView` is inside the `view`, but that means a cursor that is partially off screen is probably not going to be refreshed.

Reply:
Hmmmmm..

Would this come up when the cursor is on a double-width glyph and the viewport is scrolled such that it cuts that glyph in half?

Reply:
Yeah. Also on a double-width line, or even a double-width glyph on a double-width line, which can be four characters wide.